calculate_trigger.py

```diff
@@ -52,9 +74,47 @@ def search_for_fault_location(
         goldenrun_tb_info.at[idtbinfo, "assembler"],
         goldenrun_tb_info.at[idtbinfo, "id"],
     )
+    return [idx, ins]
+
+
+def search_for_fault_location(
+    filter_lists,
+    trigger_position,
+    fault_address,
+    trigger_occurrences,
+    fault_lifespan,
+    goldenrun_tb_exec,
+    goldenrun_tb_info,
+):
+    logger.info(f"Search trigger to fault INSN at 0x{fault_address:08x}")
+    [idx, ins] = find_fault(
+        fault_address, goldenrun_tb_exec, goldenrun_tb_info, trigger_occurrences
+    )
+    if idx < 0:
+        return [-1, trigger_occurrences, fault_lifespan]
     trigger_not_in_same_tb = 0
+    lifespan_diff = trigger_position + fault_lifespan
     trigger_position = trigger_position * (-1)
     while trigger_position != 0:
+        if idx < 0:
+            if fault_lifespan > 0:
+                if lifespan_diff < 0:
```

Comment:
These nested if-statements are hard to read.

Maybe change to something like:
```
if idx >= 0:
  return [...]
...
```

Reply:
I've been thinking about this a bit more. It is actually ok, if `lifespan_diff` is negative. I won't fault the fault address, but it will inject a fault. It is only problematic, if that causes the fault to be shifted before start. So I think, we only need to warn here, if the adjusted fault lifespan (so line 116) is smaller or equal to 0. What do you think?
This way, we also get rid of the nested if statements.

Reply:
I think we should throw away these faults with a warning, because they do not make sense at all. Why would i want to fault an instruction that is then replaced by the original instruction anyway before execution? We could check this at the beginning of the function and terminate any further calculation if the difference is not equal or greater to zero. This would resolve the problem of people having injected the fault but not seeing any changes, like it happened in #18. Also would get ride of the nested if statements

Reply:
Yep, this only works in the other fault models (for example to modify a register), but those faults should probably always be triggered correctly and not like that.
So yeah, I agree, let's just throw an error if we have a fault like that. Please add the check in a separate commit. The check can probably already be done in `calculate_trigger_addresses`.